### PR TITLE
feat: add blocto wagmi connector

### PR DESCRIPTION
**Pull request by q20274982**
## Summary

- add adapter repo `wagmi-connector`
- create `BloctoConnector` class extend from wagmi `Connector`
- add `bloctoWeb3ModalConfig ` use for web3modal

## Related Links

**Asana**:  https://app.asana.com/0/1201812548509877/1203670481412050/f

**Mockup**:  

## Checklist

- [x] Pasted Asana or Mockup link.
- [x] Tagged labels.

## Screenshot/Gif

image

### CODE REVIEW

**Comment by akira02** [comment]
And don't forgot add changeset using `yarn changeset`

**Comment by q20274982** [comment]
> The icon looks weird. Can you use the same icon with Blocto app?

image

File: .changeset/real-colts-nail.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@blocto/blocto-wagmi-connector': major
```
Comment (scottphc) [inline .changeset/real-colts-nail.md:2]: should be `minor`?

Comment (akira02) [reply .changeset/real-colts-nail.md:2]: same opinion. 
FYI redo just delete this file and add changeset again.

Comment (q20274982) [reply .changeset/real-colts-nail.md]: is able to replace the string `major` to `minor` ?

File: adapters/wagmi-connector/src/bloctoConnector.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    const provider = await this.getProvider();
+    await provider?.request({ method: "wallet_disconnect" });
+  }
+  
+  async getAccount(): Promise<`0x${string}`> {
+    const provider = await this.getProvider();
+
+    return provider
+      .request({ method: 'eth_requestAccounts' })
+      .then((accounts: string[]): `0x${string}` => accounts[0] as `0x{string}`)
```
Comment (akira02) [inline adapters/wagmi-connector/src/bloctoConnector.ts:65]: Should be as `0x${string}`

File: adapters/wagmi-connector/src/bloctoConnector.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  protected onAccountsChanged(accounts: `0x${string}`[]): void {
+    // not supported yet
+  }
+  protected onChainChanged(chain: string | number): void {
+    // not supported yet
+  }
+  protected onDisconnect(error: Error): void {
+    // not supported yet
```
Comment (akira02) [inline adapters/wagmi-connector/src/bloctoConnector.ts:127]: We support. It's `ethereum.handleDisconnect()`

**Review by akira02 — COMMENTED** [review]
And please format with `yarn format`

File: adapters/wagmi-connector/src/bloctoConnector.ts (hunk 2 above)
Comment (q20274982) [reply adapters/wagmi-connector/src/bloctoConnector.ts]: [fix(BloctoConnector): correct getAccount() return type](https://github.com/portto/blocto-sdk/pull/71/commits/1b5280823dc6807e25e963ac918a24ea87d28c59) resolved.

**Review by akira02 — APPROVED** [review]

**Pull request merged**